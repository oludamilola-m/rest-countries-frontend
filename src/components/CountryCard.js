import React from "react";
import { Link } from "react-router-dom";

const CountryCard = ({ country }) => {
  if (country) {
    return (
      <Link to={`/${country.alpha3Code}`}>
        <div className="country">
          <div className="country-flag">
            <img
              src={country.flag}
              alt="country Flag"
              className="country-img"
              data-testid="country-flag"
            />
          </div>
          <div className="country__info">
            <p className="country-name" data-testid="country-name">
              {country.name}
            </p>
            <p className="country__info-p" data-testid="country-population">
              <strong>Population:</strong> {country.population}
            </p>
            <p className="country__info-p" data-testid="country-region">
              <strong>Region:</strong> {country.region}
            </p>
            <p className="country__info-p" data-testid="country-capital">
              <strong>Capital:</strong> {country.capital}
            </p>
          </div>
        </div>
      </Link>
    );
  } else {
    return <div></div>;
  }
};

export default CountryCard;
