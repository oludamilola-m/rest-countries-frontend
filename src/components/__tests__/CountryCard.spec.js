import React from "react";
import CountryCard from "../CountryCard";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const country = {
  name: "Nigeria",
  population: "1000000",
  region: "Africa",
  capital: "Abuja",
  flag: "https://restcountries.eu/data/dza.svg",
  // alpha3Code: "DZA",
};

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<CountryCard />, div);
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <CountryCard country={country} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("displays country properties", () => {
  const { getByTestId } = render(
    <Router>
      <CountryCard country={country} />
    </Router>
  );

  expect(getByTestId("country-name")).toHaveTextContent(country.name);
  expect(getByTestId("country-flag")).toHaveAttribute("src", country.flag);
  expect(getByTestId("country-population")).toHaveTextContent(
    country.population
  );
  expect(getByTestId("country-capital")).toHaveTextContent(country.capital);
});
