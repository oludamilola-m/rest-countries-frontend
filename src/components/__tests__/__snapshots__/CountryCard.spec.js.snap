// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  href="/undefined"
  onClick={[Function]}
>
  <div
    className="country"
  >
    <div
      className="country-flag"
    >
      <img
        alt="country Flag"
        className="country-img"
        data-testid="country-flag"
        src="https://restcountries.eu/data/dza.svg"
      />
    </div>
    <div
      className="country__info"
    >
      <p
        className="country-name"
        data-testid="country-name"
      >
        Nigeria
      </p>
      <p
        className="country__info-p"
        data-testid="country-population"
      >
        <strong>
          Population:
        </strong>
         
        1000000
      </p>
      <p
        className="country__info-p"
        data-testid="country-region"
      >
        <strong>
          Region:
        </strong>
         
        Africa
      </p>
      <p
        className="country__info-p"
        data-testid="country-capital"
      >
        <strong>
          Capital:
        </strong>
         
        Abuja
      </p>
    </div>
  </div>
</a>
`;
